Fix notebook bugs for branch r1.8.0

yidong72 opened the pull request:
# What does this PR do ?

Fixed the biomegatron model NGC link error.
`get_lm_model` can get biomegatron model through model names.
Fix the biomegatron notebooks to directly load the biomegatron from NGC.
Fix the MegatronLM ckpt conversion error. 
Fix the typos in synthetic data generation notebook.

ericharper commented on examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py:
```diff
@@ -360,7 +360,7 @@ def convert(local_rank, rank, world_size, args):
 
     # TODO: reach out to PTL For an API-safe local rank override
-    trainer.accelerator.training_type_plugin._local_rank = local_rank
+    # trainer.accelerator.training_type_plugin._local_rank = local_rank
```
Is this not needed anymore? Let's delete the comment.

yidong72 replied:
deleted.

ericharper commented on nemo/collections/nlp/models/nlp_model.py:
```diff
@@ -76,20 +76,26 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None, no_lm_init=False):
             if cfg.get('language_model.config_file'):
                 config_file = self.register_artifact('language_model.config_file', cfg.language_model.config_file)
-            self.bert_model = get_lm_model(
+            bert_model = get_lm_model(
```
Shouldn't this be under a conditional?

borisfom reviewed (changes requested):
This PR rolls back many of [@Ramanathan Arunachalam](https://nvidia.slack.com/team/U020UN35TGF)’s functional changes - probably because of bad merge. 
Please examine diffs and do 'git fetch origin; git checkout -r <file>' for files you did not mean to touch. Carefully inspect other files that only your changes are include in the diff.

yidong72 commented:
> This PR rolls back many of [@Ramanathan Arunachalam](https://nvidia.slack.com/team/U020UN35TGF)’s functional changes in main (#3919) - probably because of bad merge. for example: [examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py](https://github.com/NVIDIA/NeMo/pull/3948/files#diff-c856b5e10569b76943b062527702532bb2f2034798422044086e315a61409ac4) [nemo/collections/nlp/models/nlp_model.py](https://github.com/NVIDIA/NeMo/pull/3948/files#diff-ab7ae31d24e5872426961bd8ef793bed63a18c9f4473bac372e00b1a07a9e36c)]
> 
> Please examine diffs and do 'git fetch origin; git checkout -r ' for files you did not mean to touch. Carefully inspect other files that only your changes are include in the diff.

[examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py](https://github.com/NVIDIA/NeMo/pull/3948/files#diff-c856b5e10569b76943b062527702532bb2f2034798422044086e315a61409ac4) is correct. 
I changed the `nlp_model` and `lm_util` to be compatible with the [@Ramanathan Arunachalam's](https://nvidia.slack.com/team/U020UN35TGF) changes. A lot of the changes are made to allow MegatronBERT model load from NGC. 
I also added a Jenkins test to cover the biomegatron fine-tuning.
Please help to review it.

ericharper reviewed (approved):
LGTM. Thanks!
